kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -210,27 +251,31 @@ private AtomicWriteResult flush(final long offset, final UUID txnid) {
         txnid
     );
 
-    final var writer = new AtomicWriter<K>(
+    final var writers = new HashMap<Integer, AtomicWriter<K>>();
+    subPartitioner.all(partition).forEach(p -> writers.put(p, new AtomicWriter<>(
         client,
         tableName,
-        partition,
+        p,
         plugin,
         offset,
         txnid,
         MAX_BATCH_SIZE
+    )));
+
+    buffer.values().forEach(val ->
+        writers
+            .get(subPartitioner.partition(partition, plugin.bytes(val.key)))
+            .add(val)
     );
 
-    writer.addAll(buffer.values());
+    var awr = drain(writers.values(), AtomicWriter::flush);
```

Comment:
`var awr` sounds like the talking noise Luna makes when she wants to go for a walk 😂

Reply:
(Just as a general point going forward, can we try to only use `var` when it's explicitly clear what the type is, eg `var luna = new Dog`?)

Reply:
😆 haha fair enough, I may be abusing a new language feature i'm excited about